cloudferrylib/os/compute/nova_compute.py

```diff
@@ -811,6 +818,43 @@ def update_default_quotas(self, quota_items):
 
         return self.nova_client.quota_classes.update('default', **quota_items)
 
+    def get_host_aggregates(self):
+        return self.nova_client.aggregates.list()
+
+    def _delete_host_aggregate(self, aggregate):
+        return self.nova_client.aggregates.delete(aggregate)
+
+    def deploy_host_aggregates(self, src_host_aggregates):
+        dst_host_aggregates = self.get_host_aggregates()
+        for src_aggregate in src_host_aggregates:
+            LOG.debug("Source HOST aggregates: %s", pprint.pformat(src_aggregate.to_dict()))
+            skip = False
+            for dst_aggregate in dst_host_aggregates:
+                LOG.debug("Destination HOST aggregates: %s", pprint.pformat(dst_aggregate.to_dict()))
+                if src_aggregate.name == dst_aggregate.name:
+                    if src_aggregate.metadata == dst_aggregate.metadata:
+                        LOG.info("skipping matching aggregate in destination: %s", src_aggregate.name)
+                        skip = True
+                    else:
+                        LOG.info("deleting aggregate collision in destination: %s", dst_aggregate.name)
+                        self._delete_host_aggregate(dst_aggregate)
+            if not skip:
+                self._deploy_host_aggregate(src_aggregate)
+
+    def _deploy_host_aggregate(self, aggregate):
+        LOG.info("Deploying aggregate to destination: %s", aggregate.name)
+        new_aggregate = self.nova_client.aggregates.create(aggregate.name, aggregate.metadata["availability_zone"])
+        new_aggregate = self.nova_client.aggregates.set_metadata(new_aggregate, aggregate.metadata)
+        #   adding hosts of the same name from source to destination cloud is almost never going to be legitimate
+        #   instead, need to create a host mapping convention from src_host to dst_host
+        #   until then, hosts are not added to aggregates
+        #for host in aggregate.hosts:
+        #    new_aggregate = self.nova_client.aggregates.add_host(new_aggregate, host)
+        if len(aggregate.hosts) > 0:
+            LOG.warning("Host aggregate: %s contains hosts: %s.", aggregate.name, aggregate.hosts)
+            LOG.warning("You must manually add the proper hosts in the destination cloud to this aggregate.")
+        return new_aggregate
```

Comment:
Well host aggregate seem to be a separate entity (class), so would be better to follow [separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns) principle and move this logic into a separate class covered with unit tests instead of building a [blob object](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=0CB4QFjAAahUKEwjRvqSW_-LIAhVIw2MKHcc3Bgo&url=https%3A%2F%2Fsourcemaking.com%2Fantipatterns%2Fthe-blob&usg=AFQjCNHq_tatU_75AaDzkAAqbX_Z-N6dxg) (current implementation of `NovaCompute` class)